persistence-modules/read-only-transactions/src/main/java/com/baeldung/read_only_transactions/mysql/spring/CustomSimpleJpaRepository.java

```diff
@@ -0,0 +1,32 @@
+package com.baeldung.read_only_transactions.mysql.spring;
+
+import org.hibernate.CacheMode;
+import org.hibernate.Session;
+import org.springframework.data.jpa.repository.support.JpaEntityInformation;
+import org.springframework.data.jpa.repository.support.SimpleJpaRepository;
+
+import javax.persistence.EntityManager;
+
+public class CustomSimpleJpaRepository<T, ID> extends SimpleJpaRepository<T, ID> {
```

Comment:
I think the convention says that the generic has to match the regular something like ^[A-Z][0-9]?$, that is, a letter and optionally a number (and I understand you are trying to make it meaningful).